app/src/main/java/com/geeksville/mesh/ui/ChannelFragment.kt

```diff
@@ -547,6 +565,157 @@ private fun ChannelSelection(
+                    Text(
+                        style = MaterialTheme.typography.body1,
+                        text = stringResource(id = R.string.scanned_channels)
+                    )
+                }
+                itemsIndexed(incomingChannelSet.settingsList) { index, channel ->
+                    ChannelSelection(
+                        index = index,
+                        title = channel.name.ifEmpty { modemPresetName },
+                        enabled = true,
+                        isSelected = channelSelections[index],
+                        onSelected = { channelSelections[index] = it }
+                    )
+                }
+
+                /* Current ChannelSet */
+                item {
+                    Text(
+                        style = MaterialTheme.typography.body1,
+                        text = stringResource(id = R.string.current_channels)
+                    )
+                }
+                itemsIndexed(currentChannelSet.settingsList) { index, channel ->
+                    ChannelCard(
+                        index = index,
+                        title = channel.name.ifEmpty { modemPresetName },
+                        enabled = true,
+                        onEditClick = { /* Currently we don't enable editing from the dialog. */ },
+                        onDeleteClick = {
+                            val list = currentChannelSet.settingsList.toMutableList()
+                            list.removeAt(index)
+                            currentChannelSet = currentChannelSet.copy {
+                                settings.clear()
+                                settings.addAll(list)
+                            }
+                        }
+                    )
+                }
+
+                /* User Actions via buttons */
+                item {
+                    Row(
+                        modifier = Modifier.fillMaxWidth(),
+                        horizontalArrangement = Arrangement.SpaceEvenly
+                    ) {
+                        /* Cancel */
+                        Button(
+                            onClick = onDismissRequest,
+                            modifier = Modifier
+                                .fillMaxWidth()
+                                .height(48.dp)
+                                .weight(1f)
+                                .padding(3.dp)
+                        ) {
+                            Text(
+                                style = MaterialTheme.typography.body1,
+                                text = stringResource(id = R.string.cancel)
+                            )
+                        }
+
+                        /* Replace - Replaces the previous set with the scanned channel set */
+                        Button(
+                            onClick = {
+                                onDismissRequest.invoke()
+                                viewModel.setRequestChannelUrl(Uri.parse(scannedQR))},
```

Comment:
`setRequestChannelUrl()` calls another dialog after this one (ie. the "old" dialog we're replacing), you can use `installSettings()` instead.